app/uk/gov/hmrc/breathingspaceifproxy/controller/RequestValidation.scala

```diff
@@ -48,12 +45,41 @@ trait RequestValidation extends BaseController with Logging {
       .andThen(validateStaffPidForRequestType)
   }
 
-  def validateBody[A, B](f: A => Validation[B])(implicit request: Request[AnyContent], reads: Reads[A]): Validation[B] =
-    if (!request.hasBody) Error(MISSING_BODY).invalidNec
+  def validateNino(maybeNino: String): Validation[Nino] =
+    if (DomainNino.isValid(maybeNino)) Nino(maybeNino).validNec
+    else ErrorItem(INVALID_NINO, s"($maybeNino)".some).invalidNec
+
+  def validateNino(maybeNino: Option[String]): Validation[Nino] =
+    maybeNino.fold(ErrorItem(MISSING_NINO).invalidNec[Nino])(validateNino(_))
+
+  def validateJsValue[T](json: JsValue, name: String)(implicit rds: Reads[T]): Option[T] =
+    (json \ name).validate[T] match {
+      case JsSuccess(value, _) => value.some
+      case JsError(_) => none
+    }
+
+  def validateJsObject[T](json: JsValue)(implicit rds: Reads[T]): Option[T] =
+    json.validate[T] match {
+      case JsSuccess(value, _) => value.some
+      case JsError(_) => none
+    }
+
+  def validateJsArray[T](json: JsArray, name: String, f: (T, Int) => Validation[T])(
+    implicit rds: Reads[T]
+  ): Validation[List[T]] =
+    if (json.value.isEmpty) ErrorItem(MISSING_PERIODS).invalidNec[List[T]]
```

Comment:
Uhm... `MISSING_PERIODS`, as error code, is a bit misleading here, since the function is generic and accordingly not specifically thought for checking an array of Periods.